src/modules/job-manager/alloc.c

```diff
@@ -276,14 +317,16 @@ static void alloc_response_cb (flux_t *h, flux_msg_handler_t *mh,
             errno = EEXIST;
             goto teardown;
         }
+        update_annotations (h, job, id, annotations);
         if (event_job_post_pack (ctx->event, job, "alloc",
-                                 "{ s:s }",
-                                 "note", note ? note : "") < 0)
+                                 "{ s:O? }",
```

Comment:
This will output a JSON `null` when `job->annotations` is `NULL`, is that what you intended? (Or was it meant to post an empty object? (Sorry if I missed something obvious)

Reply:
yeah, that was intentional.  Would not outputting an annotation be preferable?  I was sort of going off the same pattern before, where an empty `note` was output with an empty string.

Reply:
[RFC 21](https://flux-framework.readthedocs.io/projects/flux-rfc/en/latest/spec_21.html#alloc-event) says annotations is an optional key, so it should probably be omitted if empty.